### DISCUSSION TITLE
NFC tag scan issue

### DISCUSSION THREAD

**Discussion by Danahi** [custom-components/pyscript#307] [Q&A] [answered]
I am trying to read my tag, however I get an exception.
My code: 
`@event_trigger("tag_scanned", "tag_id == '<here>'")`

I pasted the UUID on location `<here>` and when scanning I get the following error log entry:
This error originated from a custom integration.

```
Logger: custom_components.pyscript.function
Source: custom_components/pyscript/function.py:430
Integration: Pyscript Python scripting (documentation, issues)
First occurred: 23:45:48 (7 occurrences)
Last logged: 23:48:24

run_coro: got exception Traceback (most recent call last): File "/config/custom_components/pyscript/eval.py", line 710, in call raise TypeError(f"{self.name}() called with unexpected keyword arguments: {unexpected}") TypeError: tagScanned() called with unexpected keyword arguments: device_id, tag_id
```
What am I doing wrong?

**Comment by craigbarratt** [comment, marked as answer]
The event trigger is trying to call the function with the parameters from the event, but your function doesn't declare those parameters.  It appears they are `device_id` and  `tag_id`.  So you should declare the function like this:
```
@event_trigger("tag_scanned", "tag_id == '<here>'")
def scanned_trigger(device_id=None, tag_id=None):
    ....
```
See the [docs](https://hacs-pyscript.readthedocs.io/en/stable/reference.html#event-trigger).

**Comment by JayNewstrom** [comment]
I updated my original answer here too: https://github.com/custom-components/pyscript/discussions/305

**Comment by craigbarratt** [comment]
@JayNewstrom - great, thanks!
